### ISSUE TITLE
Can we add assembly contig of NGS to canu input to reduce pacbio coverage requirement?

### ISSUE THREAD

**Issue by wangyugui** [marbl/canu#652]
Hi.

Can we add assembly contig of NGS to canu input to reduce pacbio coverage requirement?
most of contig of NGS are less than 100K. and we can use them as pacbio reads?

**Comment by wangyugui** [comment]
Can this way be used as a hybird method?

**Comment by skoren** [comment]
How low coverage are we talking about? Over about 15x you'd probably get a reasonable assembly from just the nanopore/pacbio data. 

I would not recommend using contigs as reads, there is really no good way to put in the contigs. You'd want to use them to correct the raw reads but there is no way in Canu to let one read have more of a "vote" than any other so the contigs will just be 1 piece of evidence to correct another read. Canu also assumes that 1x is not well supported so if only a contig had some sequence, it would get trimmed as it would be considered unsupported which you probably also don't want. You could maybe correct and trim the raw data using the sensitive parameters and then assemble the untrimmed contigs together with this data but that won't correct any mis-assemblies in your input.

So in short, no not really, I'd recommend one of the available hybrid methods or scaffolding methods to use your data.

**Comment by wangyugui** [comment]
Thanks for these info.

**Comment by wangyugui** [comment]
Hi @skoren 

we can copy the contigs into 4 copys before adding them to input?
such as conting001_copy1, conting001_copy2, conting001_copy3, conting001_copy4?

**Comment by brianwalenz** [comment]
I'm all for letting others burn CPU trying out things!  Instead of adding N copies of the contigs, shred them into pseudo-reads (of, say, 10kbp) at N coverage.  This will get around any problems of having exactly the same 'read' in the dataset (which we possibly detect and discount as being false data).

That said, the stuff that was assembled from short reads isn't exactly hard to assemble with long reads, so won't really help the assembly at all.  You'll just end up trimming off the stuff that isn't supported by the (longer) contigs - including all the bits in pacbio reads that could be used to join the contigs.  The (shorter) contigs with bits of repeats won't be used by canu, and even if you were to force them to be used, the mhap overlapper won't find many overlaps with them.

Look into ECtools (http://schatzlab.cshl.edu/data/ectools/) or Alpaca (https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-3927-8), however, both of those are basically unsupported any more because it's just easier and better to get more long reads.

**Comment by wangyugui** [comment]
thanks a lot.
